code/code/misc/limits.cc

```diff
@@ -467,6 +467,22 @@ int TPerson::hitGain()
   if (affectedBySpell(SPELL_ENLIVEN))
     gain *= 2;
 
+  // Scale regen based on the number of bloodlust stacks 
+  if (affectedBySpell(SKILL_BLOODLUST)) {
+    int mod = 1;
+    affectedData *ch_affected;
+
+    for (ch_affected = affected; ch_affected; ch_affected = ch_affected->next) {
+      if (ch_affected->type == SKILL_BLOODLUST) {
+        // set the mod and remove the affect so we can add it fresh
+	mod += ch_affected->modifier;
+	break;
+      }
+    }
+
+    gain *= 1+(mod/15);
```

Comment:
Gain and mod are integers so mod/15 is always going to be zero

Reply:
Updated again (and tested to ensure it's performing as desired).